salt/_pillar/metalk8s.py

```diff
@@ -100,12 +140,28 @@ def _load_iso_path(config_data):
 
 def ext_pillar(minion_id, pillar, bootstrap_config):
     config = _load_config(bootstrap_config)
+    if config.get('_errors'):
+        return {'_errors': config['errors']}
 
-    return {
+    result = {
         'networks': _load_networks(config),
         'metalk8s': {
             'products': _load_iso_path(config),
             'ca': _load_ca(config),
             'api_server': _load_apiserver(config),
         },
     }
+
+    if '_errors' in result['networks']:
+        result.setdefault('errors', []).append(
+            result['networks'].pop('_errors')
```

Comment:
Hmm don't know if we really want this, because here if we do for example
`salt-call pillar.get metalk8s:networks` we will just get None.
For me erros should be really "local" to the section.
Maybe you can have errors for network but not for metalk8s so you don't want `error` directly at the root.

IMO what we want to be able to do it's in our state we get a value lets say `pillar.get('networks')`. We have `_errors` here ok fail with the error message(s). (But indeed have all error at the "root" of the pillar can also help ... Hmm maybe we should have both local + aggregation at the root of the pillar)

Maybe someone else has another opinion here (cc @NicolasT @gdemonet )

(Also `_` missing here before `errors` no ?)

Reply:
For now this only goes as far as reproducing the existing `assert` control flow, so it seems strange in some places - indeed, it's better to have local errors, but the assertions used to make the whole thing crash violently before.

The local/global thing makes sense to me, I'll work it in unless I get any thoughts otherwise.

> (Also _ missing here before errors no ?)

Indeed!